src/components/shared/Select/index.tsx

```diff
@@ -1,153 +1,376 @@
-    fontColor: COLOR.GRAY3,
-    INVERTED: {
-      backgroundColor: COLOR.GRAY7,
-      borderColor: COLOR.GRAY3,
-      fontColor: COLOR.WHITE,
+  box: {
+    rootbox: {
+      backgroundColor: COLOR.WHITE,
+      border: {
+        borderColor: COLOR.GRAY59,
+        borderWidth: 2,
+      },
     },
-    DISABLED: {
-      backgroundColor: COLOR.VERYLIGHTGRAY,
-      borderColor: COLOR.LIGHTGRAY,
-      fontColor: COLOR.GRAY59,
+    text: {
+      fontColor: COLOR.DARK,
     },
   },
+  underbar: {
+    rootbox: {
+      backgroundColor: COLOR.WHITE,
+      border: {
+        borderBottomColor: COLOR.GRAY59,
+        borderBottomWidth: 2,
+      },
+    },
+    text: {
+      fontColor: COLOR.DARK,
+    },
+  },
+};
+
+interface ThemePropParams {
+  theme: ThemeEnum;
+  comp: CompEnum;
+  prop: StylePropEnum;
+}
+
+const getThemeProp = ({ theme, comp, prop }: ThemePropParams): string => {
+  return themeStylePropCollection[theme][comp][prop];
 };
 
-const Text = styled.Text<TextType>`
+const SelectContainer = styled.View`
+  z-index: 1;
+`;
+const Text = styled.Text<ThemeType>`
   font-size: 14px;
-  color: ${({ theme }): string => theme.fontColor};
+  color: ${(props): string =>
+    getThemeProp({
+      theme: props.theme,
+      comp: CompEnum.text,
+      prop: StylePropEnum.fc,
+    })};
 `;
 
-const RootSelect = styled.View`
-  background-color: orange;
-  width: 128;
-  height: 48;
-  /* flex: 1; */
+const RootSelect = styled.View<ThemeType>`
+  background-color: ${(props): string =>
+    getThemeProp({
+      theme: props.theme,
+      comp: CompEnum.rootbox,
+      prop: StylePropEnum.bc,
+    })};
+  ${(props): string =>
+    getThemeProp({
+      theme: props.theme,
+      comp: CompEnum.rootbox,
+      prop: StylePropEnum.bs,
+    })}
+  ${(props): string =>
+    getThemeProp({
+      theme: props.theme,
+      comp: CompEnum.rootbox,
+      prop: StylePropEnum.border,
+    })}
+  width: 128px;
+  height: 48px;
   flex-direction: row;
   justify-content: space-between;
   align-items: center;
   padding: 14px 6px;
 `;
 
+const SelectListView = styled.View`
+  elevation: 8;
+  shadow-color: ${COLOR.DODGERBLUE};
+  shadow-offset: {
+    width: 0;
+    height: 5;
+  }
+  shadow-opacity: 0.2;
+`;
+
+interface Item {
+  value: string;
+  text: string;
+}
+
+const SelectList = styled(FlatList as new () => FlatList<Item>)`
+  background-color: ${COLOR.WHITE};
+  position: absolute;
+  top: 100%;
+  left: 0;
+  padding-top: 8px;
+`;
+
+const ItemView = styled.TouchableOpacity<Selected>`
+  background-color: ${({ selected }: { selected: boolean }): string =>
+    selected ? COLOR.LIGHTBLUE : COLOR.WHITE};
+  width: 128px;
+  height: 32px;
+  padding: 6px;
+  justify-content: center;
+`;
+
+const ItemText = styled.Text<Selected>`
+  font-size: 14px;
+  color: ${COLOR.BLACK};
+`;
+
+interface ItemStyle {
+  list?: StyleProp<DefaultTheme>;
+  defaultItem?: StyleProp<DefaultTheme>;
+  selectedItem?: StyleProp<DefaultTheme>;
+}
```

Comment:
In general, itemStyle seems to have to be written as a style type, but the current code does not. 
It looks complex enough to write without reading the guide.
So I want to improve the props more intuitively.. 
```suggestion
interface ItemStyle {
  list?: StyleProp<DefaultTheme>;
  defaultItem?: StyleProp<DefaultTheme>;
  selectedItem?: StyleProp<DefaultTheme>;
}
interface ItemStyle { ... }  // item container 
interface itemTextStyle { ... } // item text
interface itemListStyle { ... } // items container
```

Reply:
Thanks for your review!
Yes, I wanted to give users more styling options but it can be too complex to write style. Then I will seperate **itemStyle** into **itemStyle** and **selectedItemStyle** removing **list**.